packages/core/core/src/services/document-service/repository.ts

```diff
@@ -0,0 +1,402 @@
+    // Find if document exists
+    const entryToUpdate = await strapi.db
+      .query(uid)
+      .findOne({ ...query, where: { ...queryParams?.lookup, ...query?.where, documentId } });
+
+    let updatedDraft = null;
+    if (entryToUpdate) {
+      const validData = await entityValidator.validateEntityUpdate(
+        model,
+        data,
+        {
+          isDraft: true, // Always update the draft version
+          locale: queryParams?.locale,
+        },
+        entryToUpdate
+      );
+
+      // Component handling
+      const componentData = await updateComponents(uid, entryToUpdate, validData as any);
+      const entryData = applyTransforms(
+        Object.assign(omitComponentData(model, validData), componentData),
+        { contentType: model }
+      );
+
+      updatedDraft = await strapi.db
+        .query(uid)
+        .update({ ...query, where: { id: entryToUpdate.id }, data: entryData });
+    }
+
+    if (!updatedDraft) {
+      const documentExists = await strapi.db
+        .query(contentType.uid)
+        .findOne({ where: { documentId } });
+
+      if (documentExists) {
+        updatedDraft = await create({
+          ...queryParams,
+          data: { ...queryParams.data, documentId },
+        });
+      }
+    }
+
+    if (updatedDraft && params.status === 'published') {
+      return publish(documentId, params).then((doc) => doc.versions[0]);
+    }
+
+    return updatedDraft;
+  }
+
+  async function count(params = {} as any) {
+    const query = await pipeAsync(
+      DP.defaultToDraft,
+      DP.statusToLookup,
+      i18n.defaultLocale(contentType),
+      i18n.localeToLookup(contentType),
+      transformParamsToQuery(uid)
+    )(params);
+
+    return strapi.db.query(uid).count(query);
+  }
+
+  async function publish(documentId: string, params = {} as any) {
+    const queryParams = await pipeAsync(
+      i18n.defaultLocale(contentType),
+      i18n.multiLocaleToLookup(contentType)
+    )(params);
+
+    await deleteFn(documentId, {
+      ...queryParams,
+      lookup: { ...queryParams?.lookup, publishedAt: { $ne: null } },
+    });
+
+    // Get deep populate
+    const entriesToPublish = await strapi.db?.query(uid).findMany({
+      where: {
+        ...queryParams?.lookup,
+        documentId,
+        publishedAt: null,
+      },
+      populate: getDeepPopulate(uid),
+    });
+
+    // Transform draft entry data and create published versions
+    const publishedEntries = await mapAsync(
+      entriesToPublish,
+      pipeAsync(
+        assoc('publishedAt', new Date()),
+        assoc('documentId', documentId),
+        omit('id'),
+        // Transform relations to target published versions
+        (entry) => {
+          const opts = { uid, locale: entry.locale, isDraft: false, allowMissingId: true };
+          return transformData(entry, opts);
+        },
+        // Create the published entry
+        (data) => create({ ...queryParams, data, locale: data.locale })
```

Comment:
won't this conflict with the create always creating a draft by defualt?

Reply:
definitely, I should have created in Draft It's not ready yet :)